Program.cs

```diff
@@ -1,4 +1,81 @@
-﻿using System;
-Console.WriteLine ("Hello, World!");
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for interns at Metamation, Batch - July 2023
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------------
+// Program.cs
+// STACK<T>
+// Implement a Stack<T> class using arrays as the underlying data structure. The Stack<T> should start with an initial capacity of 4 and double its capacity when needed.
+// Use the template shown below for implementation. Throw exceptions when necessary.
+// class TStack<T> { 
+// public void Push (T a) { }
+// public T Pop () { }
+// public T Peek () { }
+// public bool IsEmpty { get; }
+// }
+// --------------------------------------------------------------------------------------------
+using System;
+namespace Training {
+   #region Program ------------------------------------------------------------------------------
+   class Program {
+      #region Method ---------------------------------------------
+      static void Main () { }
+      #endregion
+   }
+   #endregion
+   #region TStack<T> ------------------------------------------------------------------------------
+   /// <summary>Create a class</summary>
+   /// <typeparam name="T">Datatype of the array</typeparam>
+   public class TStack<T> {
+      #region Constructor -------------------------------------------
+      /// <summary>Creating the constructor</summary>
+      public TStack () { }
+      #endregion
 
-   
\ No newline at end of file
+      #region Implementation ----------------------------------------

```

Comment:
Methods and Properties are not arranged in alphabetical order. This is a convention to follow for all submissions.